### DISCUSSION TITLE
Is there any option to set trace/span identifiers to use 64bit instead of 128?

### DISCUSSION THREAD

**Discussion by ffahri** [open-telemetry/opentelemetry-java#2953] [Q&A] [answered]
Hey, 
Is there any option in OpenTelemetry to make identifier bits to be generated 64bit?

**Comment by jkwatson** [comment, marked as answer]
Hi @ffahri . You can provide your own IdGenerator and configure the SDK with it, but it'll have to be 128bits to be a valid trace id. You can pad it with zeroes on the left, but internally it'll still be 128 bits.

**Reply by Oberon00** [reply]
See also https://www.w3.org/TR/trace-context/#interoperating-with-existing-systems-which-use-shorter-identifiers
